### PR TITLE
[libshortfin] Enable building docs with Sphinx

### PR DESCRIPTION
**Pull request by marbre**

### CODE REVIEW

File: libshortfin/docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Configuration file for the Sphinx documentation builder.
```
Comment (ScottTodd) [inline libshortfin/docs/conf.py:1]: add license/copyright header?

File: libshortfin/docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    "sphinx.ext.autodoc",
+]
+
+templates_path = ["_templates"]
+exclude_patterns = ["_build", "Thumbs.db", ".DS_Store"]
+
+# -- Options for HTML output -------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-html-output
+
+html_theme = "sphinx_rtd_theme"
```
Comment (ScottTodd) [inline libshortfin/docs/conf.py:35]: Did you try other themes? Wonder if there is something that AMD recommends

https://rocm.docs.amd.com/en/latest/contribute/toolchain.html points to https://sphinx-book-theme.readthedocs.io/en/latest/

File: libshortfin/docs/index.rst
Comment (ScottTodd) [inline libshortfin/docs/index.rst:1]: Could you add a README with steps for how to build the docs? Could just link to Sphinx docs if they work out of the box

File: libshortfin/docs/conf.py (hunk 1 above)
Comment (marbre) [reply libshortfin/docs/conf.py:3]: Added a copyright header to the files.

File: libshortfin/docs/index.rst
Comment (marbre) [reply libshortfin/docs/index.rst:1]: Sure, good point. Added a README file.

File: libshortfin/docs/conf.py (hunk 1 above)
Comment (ScottTodd) [reply libshortfin/docs/conf.py:3]: Can they use the standard format? https://github.com/nod-ai/SHARK-Platform/blob/19834ee1e1d476a116679afb9521ea0cccdeb308/shortfin/shortfin/framework/session.py#L1-L5

```
# Copyright 2024 Advanced Micro Devices, Inc.
#
# Licensed under the Apache License v2.0 with LLVM Exceptions.
# See https://llvm.org/LICENSE.txt for license information.
# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
```

**Review by ScottTodd — APPROVED** [review]

File: libshortfin/docs/conf.py (hunk 2 above)
Comment (marbre) [reply libshortfin/docs/conf.py:41]: No not yet. I wasn't able to figure out if there is some common theme used.

**Pull request merged**